src/Token/LcobucciFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key;
+
+if (!class_exists(Key\InMemory::class) && !class_exists(Builder::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Mercure\Token\LcobucciFactory" as the "lcobucci/jwt" package is not installed. Try running "composer require lcobucci/jwt".');
+}
+
+final class LcobucciFactory implements TokenFactoryInterface
+{
+    private $secret;
+
+    public function __construct(string $secret)
+    {
+        $this->secret = $secret;
+    }
+
+    public function create(array $publish, array $subscribe): string
+    {
+        if (class_exists(Key\InMemory::class)) {
+            return ($configuration = Configuration::forSymmetricSigner(
+                new Sha256(),
+                Key\InMemory::plainText($this->secret))
+            )
+                ->builder()
+                ->withClaim('mercure', [
+                    'publish' => $publish,
+                    'subscribe' => $subscribe,
+                ])
+                ->getToken($configuration->signer(), $configuration->signingKey())
+                ->toString();
+        }
+
+        // support "lcobucci/jwt" version 3
```

Comment:
Is this really necessary for a new feature?

Reply:
i agree, i think we can support only v4.